org.idempiere.test/src/org/idempiere/test/model/SalesOrderTest.java

```diff
@@ -1640,4 +1642,158 @@ public void testCreditCheckOrder()
 		assertTrue(info.isError(), info.getSummary());
 		assertEquals(DocAction.STATUS_Invalid, order.getDocStatus());
 	}
+	
+	/**
+	 * Test cases for Prepay Order - auto generate Shipment and Invoice
+	 */
+	@Test
+	public void testPrepayOrderAutoGenerate() {
+	    Timestamp today = TimeUtil.getDay(System.currentTimeMillis());
+	    	    
+	    // Create an order
+	    MOrder order = new MOrder(Env.getCtx(), 0, getTrxName());
+	    order.setC_DocTypeTarget_ID(MOrder.DocSubTypeSO_Prepay);
+	    order.setBPartner(MBPartner.get(Env.getCtx(), DictionaryIDs.C_BPartner.JOE_BLOCK.id));
+	    order.setDeliveryRule(MOrder.DELIVERYRULE_CompleteOrder);
+	    order.setDocStatus(DocAction.STATUS_Drafted);
+	    order.setDocAction(DocAction.ACTION_Complete);
+	    order.setDateOrdered(today);
+	    order.setDatePromised(today);
+	    order.saveEx();
+
+	    // Add an order line
+	    MOrderLine line1 = new MOrderLine(order);
+	    line1.setLine(10);
+	    line1.setProduct(MProduct.get(Env.getCtx(), DictionaryIDs.M_Product.AZALEA_BUSH.id));
+	    line1.setQty(Env.ONE);
+	    line1.setDatePromised(today);
+	    line1.saveEx();
+
+	    // Complete the order
+	    ProcessInfo info = MWorkflow.runDocumentActionWorkflow(order, DocAction.ACTION_Complete);
+	    assertFalse(info.isError(), info.getSummary());
+	    order.load(getTrxName());
+	    assertEquals(DocAction.STATUS_Completed, order.getDocStatus());
+	    order.saveEx();
+	    
+	    // Set both IsAutoGenerateInout and IsAutoGenerateInvoice
+	    MDocType doctype = new MDocType(Env.getCtx(), order.getC_DocType_ID(), getTrxName());
+	    doctype.setIsAutoGenerateInout(true);
+	    doctype.setIsAutoGenerateInvoice(true);
+	    doctype.saveEx();
+
+	    // Create the payment
+	    MPayment payment = new MPayment(Env.getCtx(), 0, getTrxName());
+	    payment.setC_Order_ID(order.getC_Order_ID());
+	    payment.setC_BPartner_ID(order.getC_BPartner_ID());
+	    payment.setPayAmt(order.getGrandTotal());
+	    payment.setC_Currency_ID(order.getC_Currency_ID());
+	    payment.setDocAction(DocAction.ACTION_Complete);
+	    payment.saveEx();
+
+	    // Complete the payment
+	    info = MWorkflow.runDocumentActionWorkflow(payment, DocAction.ACTION_Complete);
+	    assertFalse(info.isError(), info.getSummary());
+	    payment.load(getTrxName());
+	    assertEquals(DocAction.STATUS_Completed, payment.getDocStatus());
+
+	    // Check if a shipment was generated
+	    MInOut[] shipments = order.getShipments();
+	    assertTrue(shipments.length > 0, "No shipment was generated");
+
+	    // Check if an invoice was generated
+	    MInvoice[] invoices = order.getInvoices();
+	    assertTrue(invoices.length > 0, "No invoice was generated");
```

Comment:
should add the following assert:
- Invoice.IsPaid=Y
- Allocation line created for invoice against payment
- Invoice.C_Order_ID == Payment.C_Order_ID

Reply:
Asserts added